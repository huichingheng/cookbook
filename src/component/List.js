import React from "react";
import "../style/list.css";
import ListItem from "./ListItem";

const List = props => {
  const { data, handleClick, selectedRecipe } = props;
  
  return (
    <div className="list">
      {data.map(item => {
        return (
          <ListItem
            item={item}
            key={item.recipe_id}
            handleClick={handleClick}
            selectedRecipe={selectedRecipe}
          />
        );
      })}
    </div>
  );
};

export default List;
