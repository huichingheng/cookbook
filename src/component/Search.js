import React from "react";
import Button from "./Button";
import '../style/Search.css'
const INPUT_PLACEHOLDER = "Ingredients e.g. chicken, carrot, pasta...";
const SEARCH_BUTTON_TEXT = "Find Recipe";

const Search = props => {
  const { handleUserInput, handleSearchClick } = props;
  return (
    <div className="search">
      <input
        className="search-input"
        type="text"
        placeholder={INPUT_PLACEHOLDER}
        onChange={event => handleUserInput(event)}
      />
      <Button
        className="search-button"
        content={SEARCH_BUTTON_TEXT}
        handleClick={handleSearchClick}
      />
    </div>
  );
};

export default Search;
