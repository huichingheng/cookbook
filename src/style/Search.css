.search {
  background: tomato;
  padding: 6px 0;
  border-bottom: 1px solid black;
}
.search-input {
  /* background: blueviolet; */
  width: 60%;
}

.search-button {
    margin-left: 10px;
  }
.search-button:hover {
    background: whitesmoke;
    color: darkgray;
    box-shadow: blue;
    border-color: azure
  }
  
.classname:active {
  position: relative;
  top: 1px;
}
