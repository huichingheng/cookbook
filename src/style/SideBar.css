.sidebar{
    grid-column: 1/2;
    grid-row: 1/2;
    border: 1px solid black;
    height: 80vh;
    box-sizing: border-box
}

@media all and (max-width: 850px) {
    .sidebar{
      grid-column: 1/2;
      grid-row: 2/3;
    }
  }