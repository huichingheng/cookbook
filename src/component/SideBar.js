import React from "react";
import List from "./List";
import Search from "./Search";
import "../style/SideBar.css";

const SideBar = props => {
  const { data, handleClick, handleUserInput, handleSearchClick, selectedRecipe, handleKeyPress } = props;
  return (
    <div className="sidebar">
      <Search
        handleUserInput={handleUserInput}
        handleSearchClick={handleSearchClick}
        handleKeyPress={handleKeyPress}
      />
      <List data={data} handleClick={handleClick} selectedRecipe={selectedRecipe} />
    </div>
  );
};

export default SideBar;
