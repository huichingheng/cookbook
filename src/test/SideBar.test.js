import React from "react"
import SideBar from "../component/SideBar";
import List from "../component/List"
import Search from "../component/Search"
import { shallowRender } from "./helper/testHelper";
import { shallow } from "enzyme";

const wrapper = shallow(<SideBar />);

describe("renders component", () => {
  it("<Search />", () => {
    expect(wrapper.find(Search)).toHaveLength(1);
  });
  it("<List />", () => {
    expect(wrapper.find(List)).toHaveLength(1);
  });
});

describe("Snapshot", () => {
  const result = shallowRender(SideBar);
  it("should match", () => {
    expect(result.props).toMatchSnapshot();
  });
});
