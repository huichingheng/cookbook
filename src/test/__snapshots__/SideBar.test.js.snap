// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot should match 1`] = `
Object {
  "children": Array [
    <Search
      handleKeyPress={undefined}
      handleSearchClick={undefined}
      handleUserInput={undefined}
/>,
    <List
      data={undefined}
      handleClick={undefined}
      selectedRecipe={undefined}
/>,
  ],
  "className": "sidebar",
}
`;
